client/common/services/CacheService.js

```diff
@@ -5,56 +5,58 @@ angular.module('dashboard.services.Cache', [
 ])
 
 .service('CacheService', function() {
+  var memoryStorage = MemoryStorage();
 
   this.KEY_DELIMITER = '-';
 
   this.get = function(key) {
-    if(!localStorage.getItem(key)) return null;
+    if (memoryStorage.getItem(key)) return memoryStorage.getItem(key);
+    if (!localStorage.getItem(key)) return null;
     try {
-        var cached = JSON.parse(localStorage.getItem(key));
-        return cached;
-    }
-    catch (e) {
-        return null;
+      var cached = JSON.parse(localStorage.getItem(key));
+      return cached;
+    } catch(e) {
+      return null;
     }
   };
 
-  this.set = function(key,value) {
-    try{
-        localStorage.setItem(key,JSON.stringify(value));
+  this.set = function(key, value) {
+    try {
+      localStorage.setItem(key, JSON.stringify(value));
     } catch(e) {
-        this.remove(key);
+      this.remove(key);
+      if (e.code === 22) { // check for quota exceeded error code when in safari private browsing to not store wrong format of value in memoryStorage
+        memoryStorage.setItem(key, value);
+      }
     }
   };
 
   this.remove = function(key) {
     localStorage.removeItem(key);
   };
 
-  this.getKeyForAction = function(action,params) {
+  this.getKeyForAction = function(action, params) {
     var key = action.options.model + this.KEY_DELIMITER + action.route;
     if (action.options.api) key = action.options.api;
-    if(params) key += this.KEY_DELIMITER + JSON.stringify(params);
+    if (params) key += this.KEY_DELIMITER + JSON.stringify(params);
     return key;
-  }
+  };
 
   this.clear = function(model) {
     var key = model;
-    var regex = new RegExp('^'+key)
-    for(var k in localStorage)
-    {
-        if(regex.test(k))
-        {
-            this.remove(k);
-        }
+    var regex = new RegExp('^' + key);
+    for (var k in localStorage) {
```

Comment:
I'm guessing this doesn't really do what we want.

And with the addition of MemoryStorage, we'd need to iterate through MemoryStorage too.